[CAM-11086] new article about architecture components (#2747)
CenturyLinkCloud/PublicKB: Cloud Application Manager/Core Concepts/architecture-components.md

{{{
"title": "Architecture components",
"date": "05-22-2019",
"author": "Diego Sanjuan",
"attachments": [],

> "contentIsHTML": false

Reviewer: Please, add `keywords`

Author: done

}}}

**In this article:**

* [Overview](#overview)
* [Audience](#audience)
* [CAM Agents](#cam-agents)
* [Websockets](#websockets)
* [API Services](#api-services)
* [Orchestration Workflow](#orchestration-workflow)
* [Connectivity and Required Firewall Rules](#connectivity-and-required-firewall-rules)
* [Contacting Cloud Application Manager Support](#contacting-cloud-application-manager-support)

### Overview

This article is meant to assist users of Cloud Application Manager to learn about its components and interaction as follows:

> * CAM agents - functions, interactions, installation, and requirements

Reviewer: ```suggestion
* **CAM agents** - functions, interactions, installation, and requirements
```

> * Websockets - functions, interactions

Reviewer: ```suggestion
* **Websockets** - functions, interactions
```
Please, put all component names in bold in this list

* API Services - functions, interactions
* Orchestration Workflow - functions, interactions
* Connectivity and Required Firewall Rules

### Audience

All Cloud Application Manager users who want to learn about Cloud Application Manager component's architecture and its connectivity requirements.

### CAM Agents
Are responsible of execute event scripts in virtual machines deployed using script boxes and return their execution results. Also provide to backend network information about virtual machine regularly to inform user about virtual machine availability.

> Agent is bootstrapped and installed, in each virtual machine registered in CAM, making use of diverse execution mechanisms provided by public and private cloud providers. i.e. in AWS cloud init is used for download and install it.

Reviewer: Please, include a blank line to separate paragraphs

Author: done

Cloud Application Manager agent is written in python and it requires in linux machines to have python installed, and at least version 2.7.x to work properly.

> Agents require outbound connectivity to [https://cam.ctl.io](https://cam.ctl.io) to be able to call back and communicate results and request more work to be done on its virtual machine.

Reviewer: Idem, use a blank line to separate paragraphs and both before and after a title in the whole file

Author: done

Cloud Application Manager agent supports proxy configuration to call back when it's deployed in constrained/secured environments.

### Websockets
Cloud Application Manager uses websockets protocol for agent's communication and UI data interchange with Cloud Application Manager backend. They support real time notifications and data updates required by Cloud Application Manager's UI.

> ### API Services

Reviewer: Please, use a single blank line above, but also below a title

Author: done

Our services provide necessary [CAM API REST](https://www.ctl.io/api-docs/cam/) endpoints which are used by Cloud Application Manager UI when requesting operations and retrieving data items. These endpoints are also used by Cloud Application Manager command line tool [ebcli](https://www.ctl.io/knowledge-base/cloud-application-manager/tutorials/using-ebcli/) to list, provision and manage the lifecycle of your workloads based on box configuration.

### Orchestration Workflow

> We have persistent long running workflow which is responsible all cloud provider provisionning and orchestration to fulfill user requests and provide an updated workplan for CAM agents. It also takes care of provider's synchronization and recurrent or scheduled tasks.

Reviewer: ```suggestion
We have asynchronous and persistent long-running workflows which are responsible for all cloud provider provisioning and orchestration to fulfill user requests and provide an updated work plan for CAM agents. It also takes care of the provider's synchronization process and recurrent or scheduled tasks.
```

### Connectivity and required firewall rules

> Allow Income traffic to 443 port (OPEN) to your Cloud Application Manager Appliance (https) for UI, API calls and CAM agents requests

Reviewer: ```suggestion
Allow Income traffic to 443 (https) port (OPEN) to your Cloud Application Manager Data Center Edition (appliance) for UI, API calls and CAM agents requests
```
